### PR TITLE
Fix 500 on wrong pk for leaderbord

### LINKED ISSUE #47: CLI: Invalid key in the leaderboard command
When I try to use the `leaderboard` command on a key which doesn't exists, I get the following error:
`Error: Request failed: InternalServerError: 500 Server Error: Internal Server Error for url: http://substra-backend.node-1.com/objective/foo/leaderboard/?sort=desc`

Maybe we could have a more specific error explaining that the key doesn't exists?

### PR DESCRIPTION
**Pull request by Kelvin-M**
Fix #47

### CODE REVIEW

File: backend/substrapp/views/objective.py [hunk 1]
```diff
@@ -231,10 +231,9 @@ def data(self, request, *args, **kwargs):
 
     @action(detail=True, methods=['GET'])
     def leaderboard(self, request, pk):
-        validate_pk(pk)
-
         sort = request.query_params.get('sort', 'desc')
         try:
+            validate_pk(pk)
```
Comment (GuillaumeCisco) [inline backend/substrapp/views/objective.py]: `validate_pk(pk)` should be in its own try/except.

A try clause  should only try on one call.
Maybe creating a dedicated method:
```python
def method(self, request, pk):
    validate_pk(pk)
    sort = request.query_params.get('sort', 'desc')
    validate_sort(sort)
```

is better.
Then try/except on this method for handling all the possible cases.

Comment (Kelvin-M) [reply backend/substrapp/views/objective.py]: Thank you, I will update my PR :)

Comment (Kelvin-M) [reply backend/substrapp/views/objective.py]: I don't know if we need to create a dedicated method for this.
I'd rather prefer to use a try/except

File: backend/substrapp/views/objective.py [hunk 2]
```diff
@@ -246,6 +250,9 @@ def leaderboard(self, request, pk):
             })
         except LedgerError as e:
             return Response({'message': str(e.msg)}, status=e.status)
+        except Exception as e:
```
Comment (GuillaumeCisco) [inline backend/substrapp/views/objective.py]: if using one dedicated try/except, do you still need this?
If yes, using a dedicated method will be better for avoiding code repetition and keeping DRY.

Comment (Kelvin-M) [reply backend/substrapp/views/objective.py]: Updated !

**Pull request merged**